src/main/java/frc/robot/subsystems/DriveSubsystem.java

```diff
@@ -60,23 +87,47 @@ public DriveSubsystem() {
     m_rightFollowerMotor.follow(m_rightLeaderMotor);
 
     m_drive = new DifferentialDrive(m_leftLeaderMotor, m_rightLeaderMotor);
+    
+    // This code limits the motor current to 40 amps to prevent overheating
+
+    int rpm=3000;
+    int peakCurrent=70;
+    int continuousCurrent=40;
+
+    m_leftLeaderMotor.setSmartCurrentLimit(peakCurrent,continuousCurrent,rpm);
+    m_rightLeaderMotor.setSmartCurrentLimit(peakCurrent,continuousCurrent,rpm);
+    m_leftFollowerMotor.setSmartCurrentLimit(peakCurrent,continuousCurrent,rpm);
+    m_rightFollowerMotor.setSmartCurrentLimit(peakCurrent,continuousCurrent,rpm);
+
 
   }
 
   public void curvatureDrive(double moveSpeed, double rotateSpeed) {
     m_drive.curvatureDrive(moveSpeed, rotateSpeed, true);
   }
 
+  
+  public void pidControlObjects() {
```

Comment:
I don't believe you need this method. The `SparkPIDController.setReference()` already controls the motors based on the PID controller, so you don't need to explicitly call `CANSparkMax.set()` to set the motor speed.